azavea/echo-locator
# Display listing details on Marker

## Overview

Allows users to click on or hover over a map marker to see detailed information about an apartment, including listing photos, rental cost, number of bedrooms, address, and a link to further details. Markers and their corresponding pop-ups do not recenter the map in this PR and pop-ups display on both hover and click functionality for now, as discussed in the #358 conversation. Other than those changes, the content and styling of each popup currently matches the `demo` branch as best as possible while meeting the `develop` branch’s current styling.


### Demo
<img width="884" alt="Screen Shot 2021-11-24 at 4 42 47 PM" src="https://user-images.githubusercontent.com/36374797/143318627-181a0353-1f52-4990-b128-66fd39dfaf9b.png">


### Notes

Pop-up’s thumbnail carousel implements [Nuka](https://github.com/FormidableLabs/nuka-carousel). Library was chosen as it fits the basic carousel needs, allows for dynamic image height, and includes built-in type definitions. 


## Testing Instructions

 * Run `./scripts/server`
 * Log in and select a neighborhood (select 'Boston - Allston – 02134' neighborhood to test BHA listings)
 * Select 'Show BHA' or 'Show Realtor' to display markers
 * Confirm marker pop-up opens on mouse hover and closes on mouse exit
 * On marker click, confirm pop-up continues to display on mouse exit and closes when close button is selected
 * Confirm map does not recenter when markers/pop-ups are near edge of map bounds
 * Confirm all pop-up content is present and link and image carousel work as expected


Resolves #358
